### ISSUE TITLE
Request: Flag empty `setState(() {})` in Flutter

### ISSUE THREAD

**Issue by Phrohdoh** [dart-lang/linter#913]

**Comment by srawlins** [comment]
pq@ How does the linter go about judging use of APIs from specific packages? For example, could we also add lint around Mockito's APIs?

**Comment by pq** [comment]
Long ago we designed a plugin mechanism to support exactly this case but there were significant challenges (and other mounting priorities).  In the meantime, like with the SDK, we've allowed some package awareness to creep in (e.g., meta and Flutter generally).

Besides philosophical considerations, there are also logistics... @srawlins are you imagining support that could be provided w/o adding a `mockito` package dep?  (This I take to be important.)

**Comment by srawlins** [comment]
Right. I don't see why it should add a dependency on mockito. (I don't mean to derail this request for a `setState` lint; I just didn't know we had a mechanism for such things.)
